File: large_number_reduction.py
import math

def test_theory(n):
    left_side = 2 ** n
    right_side = 2 ** (n / 4)
    
    print(f"Testing n = {n}")
    print(f"Left side (2^n): {left_side}")
    print(f"Right side (2^(n/4)): {right_side}")
    
    if left_side < right_side:
        print("The theory holds: 2^n < 2^(n/4)")
    else:
        print("The theory does NOT hold: 2^n >= 2^(n/4)")

# Test for a range of n values
for n in range(1, 21):
    test_theory(n)
    print("-" * 40)

File: large_number_reduction.cpp
#include <iostream>
#include <cmath>

void testTheory(int n) {
    double leftSide = pow(2, n);
    double rightSide = pow(2, n / 4.0);

    std::cout << "Testing n = " << n << std::endl;
    std::cout << "Left side (2^n): " << leftSide << std::endl;
    std::cout << "Right side (2^(n/4)): " << rightSide << std::endl;

    if (leftSide < rightSide) {
        std::cout << "The theory holds: 2^n < 2^(n/4)" << std::endl;
    } else {
        std::cout << "The theory does NOT hold: 2^n >= 2^(n/4)" << std::endl;
    }
    std::cout << std::string(40, '-') << std::endl;
}

int main() {
    for (int n = 1; n <= 20; ++n) {
        testTheory(n);
    }
    return 0;
}
